### PR TITLE
Simplify wstring to QString conversion

### PR DESCRIPTION
**Pull request by tinyredpanda**

### CODE REVIEW

**Comment by CookiePLMonster** [comment]
@dolphin-emu-bot rebuild

**Comment by CookiePLMonster** [comment]
While you're at it, `CompatPatches.cpp:239` converts `event.name` from `wstring` to `string` and that looks fairly useless. Do you want to take a look?

**Comment by tinyredpanda** [comment]
I had a look and it seems that `fmt::format` doesn't support wide strings.
The compiler error message is
`'return': cannot convert from 'fmt::v5::basic_string_view<wchar_t>' to 'fmt::v5::internal::init<Context,fmt::v5::basic_string_view<char>,fmt::v5::internal::string_type>'`.

**Comment by CookiePLMonster** [comment]
Indeed, but if you also provide a wide format string and use `MessageBoxW` (which is preferable over `MessageBoxA` anyway) then it compiles.

**Comment by BhaaLseN** [comment]
Is this really the same? I remeber there being a reason why we needed `UTF16ToUTF8`...but that might be a legacy thing where this method didn't exist (in which case there are likely others that can also use the same treatment).

**Comment by tinyredpanda** [comment]
I already checked for other wstrings that are (indirectly) converted to QString, it's just this one.

File: Source/Core/DolphinQt/Main.cpp [hunk 1]
```diff
@@ -104,16 +102,16 @@ int main(int argc, char* argv[])
 
 #ifdef _WIN32
   // Get the default system font because Qt's way of obtaining it is outdated
-  NONCLIENTMETRICS metrics = {};
-  LOGFONT& logfont = metrics.lfMenuFont;
-  metrics.cbSize = sizeof(NONCLIENTMETRICS);
+  NONCLIENTMETRICSW metrics = {};
+  LOGFONTW& logfont = metrics.lfMenuFont;
+  metrics.cbSize = sizeof(NONCLIENTMETRICSW);
```
Comment (CookiePLMonster) [inline Source/Core/DolphinQt/Main.cpp]:
```suggestion
  metrics.cbSize = sizeof(metrics);
```

**Comment by CookiePLMonster** [comment]
@dolphin-emu-bot rebuild

**Comment by CookiePLMonster** [comment]
@dolphin-emu-bot rebuild

**Comment by CookiePLMonster** [comment]
Please squash `Format fix` with the commit which touched `fmt::format` first.

**Comment by CookiePLMonster** [comment]
@dolphin-emu-bot rebuild

**Comment by CookiePLMonster** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**